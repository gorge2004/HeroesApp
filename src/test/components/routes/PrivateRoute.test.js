import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/authContext';
import { MemoryRouter } from 'react-router-dom';
import PrivateRoute from '../../../components/routers/PrivateRoute';
const text = 'Saliendo de aqui';
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Navigate: ()=> (<span>{text}</span>)
}))
describe("testing_privateRoute", () => {
  Storage.prototype.setItem = jest.fn();

  test("should_render_correct_components", () => {
    const contextValue = {
      user: { logged: true, name: "Jorge" },
    };

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/']}>
                <PrivateRoute>
                    <h1>Private Route</h1>
                </PrivateRoute>
            </MemoryRouter>
        </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Private Route');
    expect(localStorage.setItem).toHaveBeenLastCalledWith('lastPath', '/');
  });
  test("should_not_render_correct_components", () => {
    const contextValue = {
      user: { logged: false },
    };

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/']}>
                <PrivateRoute>
                    <h1>Private Route</h1>
                </PrivateRoute>
            </MemoryRouter>
        </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text().trim()).toBe(text);
  });
});
