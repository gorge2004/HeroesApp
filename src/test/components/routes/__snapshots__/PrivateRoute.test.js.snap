// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing_privateRoute should_not_render_correct_components 1`] = `
<span>
  Saliendo de aqui
</span>
`;

exports[`testing_privateRoute should_render_correct_components 1`] = `
<h1>
  Private Route
</h1>
`;
